### ISSUE TITLE
cookie: does not provide an export named 'parse' (at jwt.js?v=c3e546d2:42:10)

### ISSUE THREAD

**Issue by OrJDev** [nextauthjs/next-auth#12142]
### Environment

```
  System:
    OS: macOS 14.0
    CPU: (8) arm64 Apple M1
    Memory: 471.48 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 19.3.0 - ~/.nvm/versions/node/v19.3.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v19.3.0/bin/yarn
    npm: 9.2.0 - ~/.nvm/versions/node/v19.3.0/bin/npm
  Browsers:
    Chrome: 130.0.6723.70
    Safari: 17.0
  npmPackages:
    @auth/core: ^0.37.2 => 0.37.2 
```

### Reproduction URL

Any Vite Project

### Describe the issue

When upgrading to the latest version of `@auth/core` -  "^0.37.2" and using a vite project
this error is thrown

```
Uncaught SyntaxError: The requested module '/_build/node_modules/.pnpm/cookie@0.7.1/node_modules/cookie/index.js?v=c3e546d2' does not provide an export named 'parse' (at jwt.js?v=c3e546d2:42:10)
```

this can be fixed by optimizing the deps:

```ts
import { defineConfig } from "@solidjs/start/config";

export default defineConfig({
  ssr: true,
  vite: {
    optimizeDeps: {
      include: ["@auth/core"],
    },
  },
});

````

however, when using the optimize deps approach this is being logged:

```
Sourcemap for "/Users/or/Desktop/dsgdsg/node_modules/.pnpm/oauth4webapi@3.1.2/node_modules/oauth4webapi/build/index.js" points to missing source files
```

import to mention, 
 - when using 0.35.0 it works and doesn't throw any error
 - when using 0.35.0 i can also use optimize deps and it will not log the above
 
 basically something is messed up with your cookie package configuration, currently i locked all of my projects at 0.35.0 but its not ideal.

### How to reproduce

Use any vite app with `@auth/core`

### Expected behavior

None of the source files should be missing.
